src/models/cost_function_timeseries.jl

```diff
@@ -0,0 +1,112 @@
+function get_variable_cost(
+    ts::IS.TimeSeriesData,
+    ::Union{MarketBidCost, ReserveDemandCurve},
```

Comment:
@daniel-thom should we use union here or separate the code?

Reply:
This feels like unnecessary specificity to me.  Are we using this to dispatch differently in other cases?